compendium/DeclarativeServices/src/manager/ComponentManagerImpl.cpp

```diff
@@ -60,7 +81,8 @@ namespace cppmicroservices
 
         ComponentManagerImpl::~ComponentManagerImpl()
         {
-            GetState()->Disable(*this);
+
+            GetState()->Disable(*this, nullptr);
```

Comment:
I think you're right here. I didn't see that disableFutures are appended to by AccumulateFuture which is called in the EnabledToDisabled transition. I guess this would also need to add some logic in the AccumulateFuture function to keep track of which asyncStarted value corresponds to each future in that list...

Reply:
Indeed - something like storing a `std::vector<std::pair<std::shared_future<void>, std::shared_ptr<std::atomic<bool>>>` instead of just the `shared_future`s?